rubin_sim/sim_archive/sim_archive.py

```diff
@@ -0,0 +1,1042 @@
+        scheduler_path = Path(local_data_dir).joinpath("scheduler.pickle.xz")
+        with lzma.open(scheduler_path, "wb", format=lzma.FORMAT_XZ) as pio:
+            pickle.dump(scheduler, pio)
+            in_files["scheduler"] = scheduler_path.as_posix()
+
+        sim_results = sim_runner(observatory, scheduler, **kwargs)
+
+        observations = sim_results[2]
+        reward_df = sim_results[3] if scheduler.keep_rewards else None
+        obs_rewards = sim_results[4] if scheduler.keep_rewards else None
+
+        data_dir = make_sim_archive_dir(
+            observations,
+            reward_df=reward_df,
+            obs_rewards=obs_rewards,
+            in_files=in_files,
+            sim_runner_kwargs=kwargs,
+            tags=tags,
+            label=label,
+            capture_env=True,
+            opsim_metadata=opsim_metadata,
+        )
+
+        if archive_uri is not None:
+            resource_path = transfer_archive_dir(data_dir.name, archive_uri)
+        else:
+            resource_path = ResourcePath(data_dir.name, forceDirctory=True)
+
+    results = sim_results + (resource_path,)
+    return results
+
+
+def compile_sim_archive_metadata_cli(*args):
+    parser = argparse.ArgumentParser(description="Create a metadata compilation HDF5 file at a URI")
+    parser.add_argument(
+        "--compilation_uri",
+        type=str,
+        default=None,
+        help="The URI of the metadata archive compilation to write. "
+        + "Defaults to compilation_metadate.h5 in the archive.",
+    )
+    parser.add_argument(
+        "--archive_base_uri",
+        type=str,
+        default="s3://rubin:rubin-scheduler-prenight/opsim/",
+        help="Base URI for the archive",
+    )
+    parser.add_argument(
+        "--append",
+        action="store_true",
+        help="Do not rebuild the whole compilation, "
+        + "but add new simulations with dates after the last current entry.",
+    )
+
+    arg_values = parser.parse_args() if len(args) == 0 else parser.parse_args(args)
+    archive_uri = arg_values.archive_base_uri
+    compilation_uri = arg_values.compilation_uri
+    append = arg_values.append
+    if compilation_uri is None:
+        compilation_resource = ResourcePath(archive_uri).join("compiled_metadata_cache.h5")
+    else:
+        compilation_resource = ResourcePath(compilation_uri)
+
+    compilation_resource = compile_sim_metadata(archive_uri, compilation_resource, append=append)
+
+
+def fetch_prenight_visits(
+    day_obs: str | None = None,
+    tags: tuple[str] = ("ideal", "nominal"),
+    max_simulation_age: int = 2,
+    archive_uri: str = "s3://rubin:rubin-scheduler-prenight/opsim/",
+    compilation_uri: str = "s3://rubin:rubin-scheduler-prenight/opsim/compiled_metadata_cache.h5",
+    **kwargs,
+) -> pd.DataFrame:
+    """Fetches visit parameters from the latest archived pre-night simulation.
+
+    Parameters
+    ----------
+    day_obs : `str` or  `None`
```

Comment:
It's possible that for this particular call, day_obs being None is not useful. 
We could make day_obs required here (and make it clear whether this is the day_obs of the simulation creation or if it's the day_obs of the simulation .. ).